### PR TITLE
Adding Clear Button within the search input.

### PR DESCRIPTION
**Pull request by lawong-d2l**
RALLY: https://rally1.rallydev.com/#/15545269080d/detail/userstory/167360790040?fdp=true

Adding clear button
Setting Focus on input after searching and clearing
Bumping version

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -120,6 +122,21 @@ export default {
...
+    searchUsers(searchBy) {
+      this.$store.dispatch('searchUsers', searchBy )
+      this.$refs.searchInput.focus()
+    },
+    toggleSearchButtons() {
+      if( !this.showClearButton ) {
+        return
+      }
+      if( this.searchBy != this.queryTerm ) {
+        this.$store.commit('SET_SHOW_SEARCH_BUTTON', 'true')
```
Comment (jbydeley-d2l) [inline src/App.vue]: Why are you using a string `'true'` instead of just `true`?

File: src/App.vue [hunk 2]
```diff
@@ -120,6 +122,21 @@ export default {
...
+      } else if(this.queryTerm != null ){
+        this.$store.commit('SET_SHOW_SEARCH_BUTTON', '')
```
Comment (jbydeley-d2l) [inline src/App.vue]: Do you mean `false` instead of `''`?

File: src/App.vue [hunk 3]
```diff
@@ -120,6 +122,21 @@ export default {
...
       this.searchBy = ''
       this.$store.dispatch('clearResults')
+      this.$refs.searchInput.focus()
+    },
+    searchUsers(searchBy) {
+      this.$store.dispatch('searchUsers', searchBy )
+      this.$refs.searchInput.focus()
+    },
+    toggleSearchButtons() {
+      if( !this.showClearButton ) {
```
Comment (jbydeley-d2l) [inline src/App.vue]: The logic in this method confuses me. It seems similar to the `searchUsers` action but ends up showing the search button. Seems like instead, we'd want a getter defined that returns a function to check if searchBy != queryTerm

```
function showSearchButton({queryTerm}) {
   return searchBy => queryTerm && queryTerm !== searchBy
}
```

That's untested but you could have that as a getter and remove the mutation and showSearchButton states and just use it in the `v-if` without an `@toggle`

Comment (lawong-d2l) [reply src/App.vue]: @jbydeley-d2l can we look at this one together?

**Pull request merged**